cmake/bazel_to_cmake.py

```diff
@@ -0,0 +1,324 @@
+    if key in select_cache:
+        select_name = select_cache[key]
+    else:
+        unique_values = list(
+            {*itertools.chain.from_iterable(dict.values())})
+        unique_values.sort()  # sorting ensures determinism, no spurious diffs
+        description = '_'.join(unique_values)
+        select_name = f'{package_prefix}_{select_index}_{description}'
+        select_name = select_name.replace('c++', 'cxx')
+        select_name = re.sub(r'[^a-zA-Z0-9]+', '_', select_name)
+        select_index = select_index + 1
+        select_cache[key] = select_name
+        generate_cmake_select(select_name, dict)
+
+    return [f'${{{select_name}}}']
+
+
+def generic_rule(rule_name, **kwargs):
+    print(f'{rule_name}(')
+    for key in kwargs.keys():
+        if type(kwargs[key]) is bool:
+            if kwargs[key]:
+                print(f'  {key.upper()}')
+                continue
+            else:
+                raise ValueError(
+                    'How do we specify FALSE boolean args in CMake?')
+        if key == 'visibility':
+            if kwargs[key] == ['//visibility:public']:
+                print(f'  PUBLIC')
+            continue
+        print(f'  {key.upper()}')
+        if type(kwargs[key]) is list:
+            for value in kwargs[key]:
+                if key == 'deps':
+                    target_name = get_cmake_dep_target_name(value)
+                    print(f'    {target_name}')
+                else:
+                    print(f'    {value}')
+        else:
+            if key == 'name':
+                target_name = get_cmake_local_target_name(kwargs[key])
+                print(f'    {target_name}')
+            else:
+                print(f'    {kwargs[key]}')
+    print(')\n')
+
+
+def cc_library(**kwargs):
+    generic_rule('ruy_cc_library', **kwargs)
+
+
+def cc_test(**kwargs):
+    generic_rule('ruy_cc_test', **kwargs)
+
+
+def cc_binary(**kwargs):
+    generic_rule('ruy_cc_binary', **kwargs)
+
+
+#
+# Program entry point.
+#
+
+
+if len(sys.argv) != 3:
+    print("Usage: bazel_to_cmake.py bazel_workspace_dir bazel_package_dir")
+    sys.exit(1)
+
+bazel_workspace_dir = sys.argv[1]
+bazel_package_dir = sys.argv[2]
+bazel_package_relative_dir = os.path.relpath(
+    bazel_package_dir, bazel_workspace_dir)
+package_prefix = bazel_package_relative_dir.replace(os.path.sep, '_')
+
+print("""# Copyright %d Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" % datetime.datetime.now().year)
```

Comment:
We have stupidly complex logic in IREE around these because you're actually not supposed to bump license years when updating a files, so really `now()` isn't right. My suggestion though:
1. call this generated code and drop the license headers or just use the spdx identifier
2. Copy the license header from the build file rather than rewriting it here

I wish we'd done one of those with IREE, but haven't gotten around to dealing with it

Reply:
thanks! went with option 1.